Objects/bytearrayobject.c

```diff
@@ -1729,6 +1729,11 @@ bytearray_extend(PyByteArrayObject *self, PyObject *iterable_of_ints)
 
     while ((item = PyIter_Next(it)) != NULL) {
         if (! _getbytevalue(item, &value)) {
+            if (PyErr_ExceptionMatches(PyExc_TypeError) && PyUnicode_Check(iterable_of_ints)) {
```

Comment:
The "can't extend" message (see issue) is created on line 709-11.  I expected the "cannot interpret at integer" message here, but cannot find it in the extend code.

Reply:
If I'm not wrong, lines 1709-1711 do not seem to be code relevant to the issue. The `TypeError` with the message "object cannot be interpreted as an integer" message is raised deeper in the call stack of the call to `_getbytevalue`. The `TypeError` is raised in `_PyNumber_Index`.

`_getbytevalue` from `Objects/bytearrayobject.c` calls `PyLong_AsLongAndOverflow` from `Objects/longobject.c`, which calls `_PyNumber_Index` from `Objects/abstract.c`.

I'm still new to this though, so feel free to correct me.

Reply:
IMHO it doesn't particularly matter where the error is generated. You can check the condition (and replace the error message) one or five levels above it, nobody cares. As long as you don't leak memory, that is. ;.)

Reply:
Okay. Maybe I'll just leave it where it is right now then.